projects/plugins/jetpack/modules/widget-visibility/widget-conditions.php

```diff
@@ -10,24 +10,95 @@ class Jetpack_Widget_Conditions {
 	static $passed_template_redirect = false;
 
 	public static function init() {
-		if ( is_admin() ) {
-			add_action( 'sidebar_admin_setup', array( __CLASS__, 'widget_admin_setup' ) );
-			add_filter( 'widget_update_callback', array( __CLASS__, 'widget_update' ), 10, 3 );
+		global $pagenow;
+
+		// Previews don't need widget editing loaded and also don't want to run the filter - it might mean the preview
+		// is empty which would be confusing.
```

Comment:
Would it make sense to specify what previews we are talking about here, since we're supporting so many different widget settings?

I think we may make things a bit more readable if we could add a comment about each condition mentioning what setting screen this manages.

Reply:
> I think we may make things a bit more readable if we could add a comment about each condition mentioning what setting screen this manages.

I've added some class comments explaining a little more about the different screens and circumstances and changed some other comments a little (including the preview one). I'm not sure if that fully addresses your point here though. Would you be able to give more detail on what could be improved? I'm eager to make this as easy to read as possible, without rehashing documentation about the widget APIs etc.

The other thing about screens is that use of the new API isn't bound by use a screen we know about, though this may be more of a theoretical issue.

Reply:
Thank you. I think it's way more readable right now. Since the feature covers so many different environments, more comments, even if that seems a bit much as you add them all, will always be helpful on the long run. It will also make things easier when we can one day stop supporting some of those legacy environments.